src/Orchard.Cms.Web/Startup.cs

```diff
@@ -48,7 +48,9 @@ public void Configure(IApplicationBuilder app, IHostingEnvironment env, ILoggerF
                 loggerFactory.AddDebug();
             }
 
-            app.UseModules();
```

Comment:
I explained it on next comment.
This is the implementation of `UseModules()`
```
public static IApplicationBuilder UseModules(this IApplicationBuilder app)
        {
            // Ensure the shell tenants are loaded when a request comes in
            // and replaces the current service provider for the tenant's one.
            app.UseMiddleware<ModularTenantContainerMiddleware>();
            app.UseMiddleware<ModularTenantRouterMiddleware>();

            return app;
        }
```
Problem is I need the new middleware registered just after ModularTenantContainerMiddleware registration. So, I replaced UseModules() per its implementation with the new middleware registered in the position I need.